RemoteEvaluationClient not properly cleaning up connections when timeout occurs

## Expected Behavior
The Remote Evaluation client should properly close the connection within the connection pool when a timeout occurs.

## Current Behavior
The connection in the connection pool is not being closed and is left in a bad state which causes subsequent calls to fail with request-sent

```python
[Experiment] Retry failed: Request-sent
[ERROR] (Amplitude) [Experiment] Retry failed: Request-sent
```

## Possible Solution
Handle timeout exceptions by setting `is_available` to `False` in order to properly clean up connections.
```python
try:
    response = conn.request('POST', '/sdk/vardata', body, headers)
    elapsed = '%.3f' % ((time.time() - start) * 1000)
    self.logger.debug(f"[Experiment] Fetch complete in {elapsed} ms")
    json_response = json.loads(response.read().decode("utf8"))
    variants = self.__parse_json_variants(json_response)
    self.logger.debug(f"[Experiment] Fetched variants: {json.dumps(variants, default=str)}")
    return variants
except TimeoutError as e:
    conn.is_available = False
    raise e
finally:
    self._connection_pool.release(conn)
```

## Steps to Reproduce
<!--- Please provide a clear sequence of steps to reproduce this bug --> 
<!--- Include code and images, if relevant -->
1.  Set the timeout to a very low value 
```python
evaluation_config = RemoteEvaluationConfig(fetch_retries=1, fetch_timeout_millis=1)
Experiment.initialize_remote(api_key, evaluation_config)
```
3. Call Amplitude 
```python
experiment.fetch_async(user, callback=callback)
```
4. Observe the connection is not disposed and as such all calls after fail
```python
[Experiment] Retry failed: Request-sent
[ERROR] (Amplitude) [Experiment] Retry failed: Request-sent
```

## Environment
- SDK Version: 1.1.1
- Language Version: 3.11.2

--- a/src/amplitude_experiment/connection_pool.py
+++ b/src/amplitude_experiment/connection_pool.py
@@ -1,5 +1,6 @@
 import threading
 import time
+import logging
 from typing import Any
 
 from http.client import HTTPConnection, HTTPResponse, HTTPSConnection
@@ -28,9 +29,13 @@ class WrapperHTTPConnection:
         self.pool.release(self)
 
     def request(self, *args: Any, **kwargs: Any) -> HTTPResponse:
-        self.conn.request(*args, **kwargs)
-        self.response = self.conn.getresponse()
-        return self.response
+        try:
+            self.conn.request(*args, **kwargs)
+            self.response = self.conn.getresponse()
+            return self.response
+        except Exception as e:
+            self.close()
+            raise e
 
     def close(self) -> None:
         self.conn.close()

